src/Framework/Constraint/IsResourceType.php

```diff
@@ -0,0 +1,60 @@
+<?php
+/*
+ * This file is part of PHPUnit.
+ *
+ * (c) Sebastian Bergmann <sebastian@phpunit.de>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+namespace PHPUnit\Framework\Constraint;
+
+use PHPUnit\Util\InvalidArgumentHelper;
+
+/**
+ * Constraint that checks if value is resource type
+ *
+ * The file path to check is passed as $other in evaluate().
+ */
+class IsResourceType extends Constraint
+{
+    /**
+     * @var string
+     */
+    private $type;
+
+    /**
+     * @param string $type
+     *
+     * @throws \PHPUnit\Framework\Exception
+     */
+    public function __construct($type)
+    {
+        parent::__construct();
+
+        if (!\is_string($type)) {
+            throw InvalidArgumentHelper::factory(1, 'string');
+        }
+
+        $this->type = $type;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    protected function matches($other)
```

Comment:
actually, my concern is still valid.
once, it was for assertion method, but assertion is only proxy for constraints. Or is it not?
I can use this newly created Constraint for non-resource. Assertion has dedicated support for it, yet constraint doesn't. I believe it shall be moved to constraint.

Reply:
Thanks again.

> I can use this newly created Constraint for non-resource.

I got the whole point now, you are right.

> Assertion has dedicated support for it, yet constraint doesn't.

right, I missed that types were introduced, great job!

> I believe it shall be moved to constraint.

Could you confirm that:
If constraint does something with specific data type, type must be validated inside.
So `IsResourceType` should be `IsResourceOfType` that uses `IsType` internally?

Reply:
> If constraint does something with specific data type, type must be validated inside.

indeed

> So IsResourceType should be IsResourceOfType that uses IsType internally?

commented separatelly